pkg/proxy/iptables/proxier.go

```diff
@@ -210,6 +212,9 @@ type Proxier struct {
 	// networkInterfacer defines an interface for several net library functions.
 	// Inject for test purpose.
 	networkInterfacer proxyutil.NetworkInterfacer
+
+	// nfAcctCounters hold NFAcct counters mapped to their existence in kernel.
+	nfAcctCounters map[string]bool
```

Comment:
I was planning to use Metric as the key for this, directly integrating metric with nfacct subsystem...
But there is a restriction of 31 characters for name of the nfacct object and our metrics names goes till 60 chars.

```go 
map[Metric]bool
```

Reply:
also, we need to have two counters - one for each stack to avoid corruption, final metric can be the sum of both.

Considering we all have to append the IP version in the counter should I hash the name and then store?

Reply:
Yeah, we should probably use hashed names.

The hashing should maybe be part of the `nfacct` package.

> also, we need to have two counters - one for each stack to avoid corruption

So originally, when we were parsing the counters out of `iptables-save` data, it made sense to have the metric updates be done as part of `syncProxyRules`, but with the nfacct-based approach, there's no real reason to read the counters there, other than "we know that this code gets run periodically". Eventually (maybe when we finish untangling the other dual-stack health/metrics stuff) we could have the "read nfacct and update metrics" code be a separate controller, and then there could just be one for the whole kube-proxy rather than one per Proxier.